docs/blueprints/deletion-performance.md

```diff
@@ -0,0 +1,39 @@
+# Deletion Performance
+
+## Abstract
+
+This document describes design for improvments to deletion speed.
+
+## Background
+
+Current design requires uplinks to send deletion information to all the storage nodes. This ends up taking a considerable amount of time.
+
+There are few goals with regards to deletions:
+
+- We need to ensure that uplinks are responsive.
+- We need to ensure that storage nodes aren't storing too much garbage, since it reduces overall capacity of the network.
+- We need to ensure that satellites aren't storing too much garbage, since that increases running cost of the satellite.
+
+TODO: where exactly are deletes spending time.
+
+## Design
+
+First we can do reduce timeouts for delete requests. Undeleted pieces will eventually get garbage collected, so we can allow some of them to get lost.
+
+The uplink should be able to issue the request without having to wait for the deletion to happen. Currently deletions is implemented as an RPC, instead, use a call without waiting for a response.

```

Comment:
I don't get this point. What "a call" means here? Doesn't an RPC response time base on what the server is executing rather than the protocol response itself?

Reply:
For example, node could internally delete things async; alternatively we could use UDP for delete requests.